client/extensions/woocommerce/state/sites/payment-methods/reducer.js

```diff
@@ -4,27 +4,37 @@
 import { createReducer } from 'state/utils';
 import { LOADING } from 'woocommerce/state/constants';
 import {
-	WOOCOMMERCE_PAYMENT_METHODS_REQUEST,
-	WOOCOMMERCE_PAYMENT_METHODS_REQUEST_SUCCESS,
+	WOOCOMMERCE_PAYMENT_METHOD_ENABLED_UPDATE,
+	WOOCOMMERCE_PAYMENT_METHOD_ENABLED_UPDATE_SUCCESS,
 	WOOCOMMERCE_PAYMENT_METHOD_UPDATE,
 	WOOCOMMERCE_PAYMENT_METHOD_UPDATE_SUCCESS,
+	WOOCOMMERCE_PAYMENT_METHODS_REQUEST,
+	WOOCOMMERCE_PAYMENT_METHODS_REQUEST_SUCCESS,
+
 } from 'woocommerce/state/action-types';
 
 // TODO: Handle error
 
 export default createReducer( {}, {
-	[ WOOCOMMERCE_PAYMENT_METHODS_REQUEST ]: () => {
-		return LOADING;
+	[ WOOCOMMERCE_PAYMENT_METHOD_ENABLED_UPDATE ]: ( state ) => {
+		// TODO: Return some sort of saving indicator
+		return state;
 	},
 
-	[ WOOCOMMERCE_PAYMENT_METHODS_REQUEST_SUCCESS ]: ( state, { data } ) => {
-		return data;
+	[ WOOCOMMERCE_PAYMENT_METHOD_ENABLED_UPDATE_SUCCESS ]: ( state, { data } ) => {
+		const methods = state || [];
+		const newMethods = methods.map( ( method ) => {
```

Comment:
IDK, I had to look at this for a little bit to understand what it was doing. Maybe a comment saying "iterate over our methods and update just the one that matches the method in the passed data" or something?